packages/core/src/validate.js

```diff
@@ -264,17 +265,25 @@ export default function validateFormData(
 }
 
 /**
- * Get a similar schema where ref's are prefixed with "__rjsf_rootSchema"
+ * Get a similar schema where ref's are prefixed with `rootSchemaId`
  * This is used in isValid to make references to the rootSchema
  */
-function withIdRefPrefix(schema) {
-  const obj = { ...schema };
-  for (let key of Object.keys(obj)) {
-    const value = obj[key];
-    if (key === "$ref") {
-      obj[key] = "__rjsf_rootSchema" + value;
-    } else if (value.constructor === Object) {
-      obj[key] = withIdRefPrefix(value);
+function withIdRefPrefix(schemaNode) {
+  let obj = schemaNode;
```

Comment:
```suggestion
  let obj;
```

Reply:
That would break stuff. The input name may be misleading. `schemaNode` can be primitive, Object or array. It should return accordingly